src/pages/index.jsx

```diff
@@ -1,11 +1,30 @@
 import React from 'react';
 import { IntlProvider } from 'react-intl';
+import { Provider } from 'react-redux';
+import { ConnectedRouter } from 'react-router-redux';
+import { PrivateRoute } from '@edx/frontend-auth';
+
 import Layout from '../components/Layout/Layout';
 
-const IndexPage = () => (
-  <IntlProvider locale="en">
-    <Layout />
-  </IntlProvider>
+import apiClient from '../data/apiClient';
+import store from '../data/store';
+import history from '../data/history';
+
+const AppWrapper = () => (
+  <Provider store={store}>
+    <IntlProvider locale="en">
+      <Layout>
+        <ConnectedRouter history={history}>
+          <PrivateRoute
+            path="/"
+            component=""
+            authenticatedAPIClient={apiClient}
+            redirect={process.env.BASE_URL}
+          />
+        </ConnectedRouter>
+      </Layout>
+    </IntlProvider>
+  </Provider>
 );
 
-export default IndexPage;
+export default AppWrapper;
```

Comment:
Because of the way gatsby works, the `index` file in pages does not wrap the entire app, only that page. So redux will only be available for that page with this setup. Gatsby allows you to attach to the root of the project through config files (in this case `gastby-browser.js` and `gatsby-ssr.js` need to be created). 

Gatsby has a good example on how to do this: https://github.com/gatsbyjs/gatsby/tree/master/examples/using-redux

Reply:
I am getting a "WebpackError: ReferenceError: window is not defined" when I implement the above ... possibly because of 
https://github.com/edx/frontend-auth/blob/05ab8af95ef2f79a57d3f54632508d5ed8898a79/src/AuthenticatedAPIClient/authInterface.js#L57

I have tried https://www.gatsbyjs.org/docs/debugging-html-builds/#fixing-third-party-modules
However, error persists ...

Reply:
This is an error that has come up a few times recently. It's been fixed by adding `globalObject: 'typeof self !== \'undefined\' ? self : this',` to the offending packages `webpack.config.js`.

here's the fix as we used it in the header as an example: https://github.com/edx/frontend-component-site-header/pull/18/files

Hopefully that will fix it in this case too.

Reply:
No luck ... apparently gatsby-config.js does not support "output" ... my config:
```
module.exports = {
  siteMetadata: {
    logo: 'https://www.edx.org/sites/default/files/open-edx-logo-with-reg.png',
    siteName: 'Open Edx',
  },
  output: {
    filename: 'main.js',
    library: 'frontend-auth',
    libraryTarget: 'umd',
    globalObject: 'typeof self !== \'undefined\' ? self : this',
  },
  plugins: [
    {
      resolve: 'gatsby-plugin-sass',
      options: {
        includePaths: [
          `${__dirname}/node_modules`,
        ],
      },
    },
  ],
};
```
https://www.gatsbyjs.org/docs/gatsby-config/

Reply:
Sorry for the confusion. I meant to say that line needs to go into the `@edx/front-auth` package, not this repo. So a pull request with the webpack changes needs to be made. No webpack changes to gatsby need to be made here.

Reply:
Ah! ... got it ... thanks!

Reply:
Added a PR https://github.com/edx/frontend-auth/pull/24 for fixing the window issue

The PR above did not solve the issue.

The 'window' check at 
https://www.gatsbyjs.org/docs/debugging-html-builds/#fixing-third-party-modules was locally deployed inside frontend-auth at 
https://github.com/edx/frontend-auth/blob/05ab8af95ef2f79a57d3f54632508d5ed8898a79/src/AuthenticatedAPIClient/authInterface.js#L57

While the above removed the window issue, following error has appeared:
WebpackError: ReferenceError: regeneratorRuntime is not defined

Reply:
I have moved the Provider to Layout ... which seems to work

Reply:
Does it build without error if you run `npm run build`? the travis logs are still showing a `window is not defined` error (Even though it still passed the check, which is something I'm going to look at now). https://travis-ci.org/edx/frontend-app-learner-portal/jobs/526444244

I'd rather not have the `<Provider>` in `<Layout>`, but if it builds, we can roll with it for now. But we should fix this in the future.

Reply:
@irfanuddinahmad I'd say hold off on this for now. Some research shows us this problem won't have an easy solution, so we're deciding where to go from here.

Reply:
@irfanuddinahmad Yeah, I agree that we should avoid putting the `<Provider>` in `<Layout>`.

We may need to figure out a way to make the @edx/frontend-auth package compatible with server-side rendered MFEs, or piece together our own solution. Team Aperture is doing some exploration around this, so let's plan on holding off on this for the time being.